front/src/applications/rollingStockEditor/components/RollingStockEditorCurves.tsx

```
@@ -0,0 +1,146 @@
+
+    const filledMatrix: (
+      | {
+          value: string;
+        }
+      | undefined
+    )[][] =
+      speedsList !== undefined && effortsList !== undefined
+        ? effortsList.map((effort, index) => [
+            { value: (speedsList[index] * 3.6).toString() },
+            { value: (effort / 10).toString() },
+          ])
+        : [];
+
+    if (isEmpty(filledMatrix)) return EMPTY_MATRIX;
+
+    return !isEmpty(filledMatrix) && filledMatrix.length > 8
+      ? filledMatrix
+      : filledMatrix.concat(createEmptyMatrix<CellBase<string>>(8 - filledMatrix.length, 2));
+  };
+
+  const [curves, setCurves] = useState<Matrix<CellBase<string>>>(
+    defaultCurve(currentRsEffortCurve)
+  );
+
+  const updateSpreadsheet = (e: Matrix<{ value: string }>) => {
+    const emptyRow = [undefined, undefined];
+    const sheetValues = e.filter(
+      (step) => step[0]?.value !== undefined || step[1]?.value !== undefined
+    );
+
+    if (!isEmpty(sheetValues) && sheetValues.length >= 8) {
+      sheetValues.push(emptyRow);
+      setCurves(sheetValues);
+    } else if (!isEmpty(sheetValues) && sheetValues.length < 8) {
+      const fillingRows = createEmptyMatrix<CellBase<string>>(8 - sheetValues.length, 2);
+      setCurves(sheetValues.concat(fillingRows));
+    } else {
+      setCurves(defaultCurve(currentRsEffortCurve));
+    }
+  };
+
+  const parseCurve = (e: Matrix<{ value: string }>) => {
+    const sheetValues = e
+      .filter(
+        (step) =>
+          step[0]?.value !== (undefined && /^\s+$/) && step[1]?.value !== (undefined && /^\s+$/)
+      )
+      .map((row) => row.map((rowValue) => rowValue?.value.replaceAll(',', '.')));
+
+    // --- Partie pour mettre à jour le state avec les données de la nouvelle courbe ---
+    const newCurve: Curve = { max_efforts: [], speeds: [] };
+    if (!isEmpty(sheetValues)) {
+      sheetValues.forEach((step) => {
+        newCurve.speeds.push(Number(step[0] as string) / 3.6);
```

if `stop[0]` is `null` of `undefined` `Number(stop[0])` is gonna be `NaN`.

- if we’re sure the step is never `undefined`, use `string` instead of `string | undefined`. 
- if it can be undefined,  handle the case with  (`push(0)` in that case for example).

`as` should not be used as a way to force TS to hide the error.

---

Done